site/app/controllers/forum/ForumController.php

```diff
@@ -916,6 +916,18 @@ public function getThreads($page_number = null) {
      * @Route("/courses/{_semester}/{_course}/forum/threads/single", methods={"POST"})
      */
     public function getSingleThread() {
+        // Checks if thread id is null. If so, render "fail" json response case informing that thread id is null.
+        if ($_POST['thread_id'] == null) {
+            return $this->core->getOutput()->renderJsonFail("Invalid thread id (NULL ID)", $_POST['thread_id']);
+        }
+        // Checks if thread id is not an integer value. If so, render "fail" json response case informing that thread id is not an integer value.
+        if (!ctype_digit($_POST['thread_id'])) {
```

Comment:
Should there be !(is_int($_POST['thread_id']) || ctype_digit($_POST['thread_id'])) ?